server/src/main/java/org/elasticsearch/search/sort/ScriptSortBuilder.java

```diff
@@ -246,10 +248,15 @@ public static ScriptSortBuilder fromXContent(XContentParser parser, String eleme
 
     @Override
     public SortFieldAndFormat build(SearchExecutionContext context) throws IOException {
-        return new SortFieldAndFormat(
-            new SortField("_script", fieldComparatorSource(context), order == SortOrder.DESC),
-            DocValueFormat.RAW
-        );
+        DocValueFormat docValueFormat = DocValueFormat.RAW;
+        if ("version".equals(this.type.toString())) {
+            MappedFieldType fieldType = context.getFieldType("version");
```

Comment:
this is wrong... I thought it returned a field based on the type name, but it's based on the *field* name.